### DISCUSSION TITLE
Pivot or Model or both?

### DISCUSSION THREAD

**Discussion by SledgehammerPL** [wintercms/winter#750] [Q&A] [answered]
I want to use WinterCMS for managing - so the most important part is backend not frontend (for now)

I have models:

member
id|name

fee
id| fee_type

memberfee (for now it is pivot)
member_id|fee_id|date|value|description

And now I realized that I need backend-menu - Member Fees. 
The obvious solution is to add controller MemberFees and attach it to backend Menu button.

But
I have already memberfeepivot.

And now - should I change memberfee class from pivot to model (and then use it in controller) or should I create model beside of pivot, or maybe something else?

**Comment by mjauvin** [comment]
Not sure I understand 100%, but I would create a dropdown field in your pivot and assign it options using getFieldOptions() and select those when creating the relation pivot data.

That `getFieldOptions()` method could fetch its values from another model where you have a controller... you could also use a field of type "relation" in the pivot data, but not sure if it's fully supported (maybe try this first).

**Reply by SledgehammerPL** [reply]
No, I want to have  backend columnlist with fees paid by Members.

(I want to use WinterCMS for managing - so the most important part is backend not frontend (for now)

**Comment by mjauvin** [comment, marked as answer]
You can always add an action to your Members controller to show the fees paid. Otherwise, the Pivot class actually extends Model, so you could probably add a Controller for it, I guess.

**Reply by SledgehammerPL** [reply]
If Pivot extends Model - I have no more questions :D

**Reply by SledgehammerPL** [reply]
(making controller for pivot with builder is quite hard - but works :D
